src/main/java/org/jabref/model/groups/WordKeywordGroup.java

```diff
@@ -113,6 +118,14 @@ public boolean contains(BibEntry entry) {
 
     private Set<String> getFieldContentAsWords(BibEntry entry) {
         if (onlySplitWordsAtSeparator) {
+            if (BibEntry.TYPE_HEADER.equals(searchField)) {
+                for (String searchWord : searchWords) {
+                    Optional<EntryType> entryType = EntryTypes.getType(entry.getType(), BibDatabaseMode.BIBLATEX);
```

Comment:
I think you can move the EntryType.get... Before the loop

Reply:
Then you could implement the for loop as stream with filter and in the end you can use map / and collectors.toSet

Reply:
Thanks. I have a question - While fetching entryType, how do I decide which BibDatabaseMode to pass? BIBTEX or BIBLATEX? In both the committed files I have used BIBLATEX. Is that correct?